### ISSUE TITLE
Intl.Segmenter TypeScript error

### ISSUE THREAD

**Issue by GJZwiers** [denoland/deno#14182]
see: https://stackoverflow.com/questions/71715016/intl-segmenter-support-in-deno

The following runs OK in JavaScript:

```js
const SEGMENTER = new Intl.Segmenter('en', { granularity: 'grapheme' });
```

but throws in TypeScript, unless `--no-check` is used:

```
error: TS2339 [ERROR]: Property 'Segmenter' does not exist on type 'typeof Intl'.
const SEGMENTER = new Intl.Segmenter('en', { granularity: 'grapheme' });
```

Similar results for the [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter) example,:

```ts
const segmenterFr = new Intl.Segmenter('fr', { granularity: 'word' });
const string1 = 'Que ma joie demeure';

const iterator1 = segmenterFr.segment(string1)[Symbol.iterator]();

console.log(iterator1.next().value.segment);
// expected output: 'Que'

console.log(iterator1.next().value.segment);
// expected output: ' '

```

**Comment by petamoriken** [comment]
The type definition file is not yet prepared on the upstream side.

**Comment by petamoriken** [comment]
Reported to TypeScript https://github.com/microsoft/TypeScript/issues/48523

**Comment by petamoriken** [comment]
Merged in TypeScript
https://github.com/microsoft/TypeScript/pull/48800

**Comment by petamoriken** [comment]
This issue will be fixed by #14242

**Comment by GJZwiers** [comment]
Closing because the error has been fixed in v1.23. Thanks @petamoriken for reporting the issue to TypeScript.
